oarepo_vocabularies/authorities/providers/openaire_provider.py

```diff
@@ -0,0 +1,255 @@
+import base64
+import logging
+import os
+from flask import current_app
+import requests
+
+from oarepo_vocabularies.authorities.providers.base import AuthorityProvider
+
+
+logger = logging.getLogger("oarepo-vocabularies.providers.openaire")
+
+class OpenAIREClient(object):
+    
+    def __init__(self, client_id, client_secret, url=None, testing=False, timeout=None, **kwargs):
+        self.client_id = client_id
+        self.client_secret = client_secret
+        self.testing = testing 
+        self.timeout = timeout or 10000
+        
+    def get_token(self):
+        url = "https://aai.openaire.eu/oidc/token"
+        credentials = f"{self.client_id}:{self.client_secret}"
+        encoded_credentials = base64.b64encode(credentials.encode('utf-8')).decode('utf-8')
+        
+        headers = {
+            "Authorization": f"Basic {encoded_credentials}"
+        }
+        
+        data = {
+            "grant_type": "client_credentials"
+        }
+        
+        try:
+            response = requests.post(url, headers=headers, data=data)
+            response.raise_for_status() 
+            return response.json().get("access_token")
+        except requests.exceptions.HTTPError as http_err:
+            print(f"HTTP error occurred: {http_err}")
+        except Exception as err:
+            print(f"Other error occurred: {err}")
+    
+    def quick_search(self, access_token, search_query="", page=1, page_size=20 ):
+        url = "https://api.openaire.eu/search/projects?format=json"
+        if not access_token:
+            return {}
+        headers = {
+            "Authorization": f"Bearer {access_token.strip()}"
+        }
+        
+        if not search_query or search_query == "":
+            return {}
+        
+        params = {
+            "name": search_query,
+            "page": page,
+            "size": page_size
+        }
+        
+        response = requests.get(url, headers=headers, params=params)
+        if response.status_code != 200:
+            print(f"Error response: {response.status_code}")
+            print(f"Response content: {response.text}")
+        response.raise_for_status()
+        return response.json()
+    
+    def get_record(self, item_id, access_token):
+        url = f"https://api.openaire.eu/search/projects?openaireProjectID={item_id}&format=json"
+        
+        headers = {
+            "Authorization": f"Bearer {access_token.strip()}"
+        }
+        
+        response = requests.get(url, headers=headers)
+        response.raise_for_status()
+        return response.json()
+            
+    
+class OpenAIREProvider(AuthorityProvider):
+    
+    def __init__(self, url=None, testing=False, **kwargs):
+        try:
+            client_id = current_app.config["OPENAIRE_CLIENT_ID"]
+            client_secret = current_app.config["OPENAIRE_CLIENT_SECRET"]
+        except RuntimeError:
+            client_id = os.environ["INVENIO_OPENAIRE_CLIENT_ID"]
```

Comment:
Assume you always have current_app and remove the os.environ code (if needed in tests, add "app" fixture to test arguments

Reply:
I assumed I should keep this since we also have it at orcid_provider.py. Will change it there too in that case.